### ISSUE TITLE
About the approval pop that appears when switching Wi-Fi

### ISSUE THREAD

**Issue by onm168** [joaomgcd/TaskerSettings#56]
Sorry for the translation. Please forgive me if there are any strange documents.
Device used is Xiaomi Redmi note 13 Pro 5G（OS:Xiaomi Hyper OS Android 14)

As usual, the approval popup keeps appearing when switching to Wifi. 
Previously, I was using Redmi Note 9S and was able to use it with Huawei and Xiaomi ver's taskersettings.

I also carefully read the story of #13.
https://github.com/joaomgcd/TaskerSettings/pull/13#issuecomment-1259512766

This is the same situation where taskersettings is not displayed.

Therefore, it is not always possible to set permissions in the OS settings. 
I think the number of people facing this problem will increase in the future, 
so I would like to help in any way I can. I will also cooperate. (I'm a user, not a developer)

**Comment by parena1979** [comment]
I am having the same issue on Nothing (2a), now on Android 15, but also before on 14. This happens with both the regular and the Xiaomi versions of Tasker Settings. The Wi-Fi Control setting allows Tasker and Tasker Settings to change Wi-Fi, which I believe is the setting that should prevent the popup.

I'm guessing this is not a Tasker Settings issue, but as there was a fix for the Xiaomi phones, perhaps there _is_ something that can be done on the Tasker (Settings) side.

**Comment by onm168** [comment]
I found it!
However, even if this setting is enabled, POPUP will still be displayed.
settings>Privacy protection>Special Permissions>Wi-fi control>taskersettings(tasker)>Allow app to control Wifi

If you use taskersettings, which is not the Xiaomi version, Bluetooth POP will also appear (of course). By using the Xiaomi compatible version, Bluetooth
POPUP can be avoided.

Is this POPUP for Wifi really special?
![1000000083](https://github.com/user-attachments/assets/aeacf884-4989-4e08-bf15-d59047607012)

**Comment by silverspear777** [comment]
I have the same issue! Xiaomi 11 ultra here.
I was previously on android 13 with no issue. but once I upgraded to hyperOS (android 14), I had this popup issue, even with the Xiaomi version of the Tasker settings. please help.

**Comment by joaomgcd** [comment]
Unfortunately I don't think I can help, sorry :( It's just how some ROMs work.

**Comment by Leseratte10** [comment]
I have the same issue on my Redmi Note 14. 

Given that the issue seems to be that that particular permission no longer shows up in the menus, maybe it can still manually be set using adb grant?

Do you know how this particular permission is named internally so I could try running "adb shell pm grant xxx yyy" and see if that helps?

**Comment by joaomgcd** [comment]
I think these should be the needed permissions:

<uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
    <uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />

**Comment by silverspear777** [comment]
I want to update... i upgraded from xiaomi 11ultra, to a xiaomi 14 ultra, both on HyperOS 1.0, and the problem disappeared. So i agree wtih the maker, that this is most likely a ROM issue.

**Comment by onm168** [comment]
It's been a while, everyone!

@All
My Xiaomi Redmi Note 13 Pro has been updated to Android 16!
And now the pop-ups have completely disappeared!
It's so strange. I can't believe that pop-up notifications disappear when the Android OS is updated!
Now I can enjoy a comfortable Tasker life for a while longer!

**Comment by joaomgcd** [comment]
That's great! Thanks for letting us know!

**Comment by kanlukasz** [comment]
`closed as completed` ? Please note that this issue affects other devices and still persists.

**Comment by parena1979** [comment]
@kanlukasz @joaomgcd  Correct, my Nothing Phone (2a) still has the exact same issue after the Android 16 update.

**Comment by joaomgcd** [comment]
Sorry, I thought it was fixed. It's not something I can do to fix anyway, so I now closed it as "won't fix". Sorry!

**Comment by gitterman** [comment]
From the many comments, it seems to be more google related than brand related. Thus, it may be advisable to give feedback to google to stop the crappy infantilization and give back full Wi-Fi control to the end users.

**Comment by joaomgcd** [comment]
@gitterman but on Google devices (like Pixel phones) the extra popup doesn't appear at all. I've never heard from anyone with a Pixel phone that has that popups. It does seem brand related unfortunately.

**Comment by onm168** [comment]
@gitterman 
Vanilla Android OS is not to blame, and since there are no issues with devices
like the Google Pixel that have a pure OS, it is wrong to sue Google. 
If that were to happen, there is a possibility that the future of Taskersettings would be threatened, 
so I think it would be better to stop.

If you're going to blame someone, blame the OS or the device itself, which is each company's own platform. That's what I think. For this kind of problem.

However, I was lucky this time, 
but if it became too much for me to use, 
I was thinking of upgrading to a Google Pixel :)
